backend/app/views.py

```diff
@@ -112,24 +114,129 @@ def search_photographers(request):
     TODO: Add pagination for both cases (when given a search query and when nothing is given) 
     so that the user is sent the first 50 results and they can view more results as they scroll down the page.
     """
+
+    def parse_order_by(order_by):
+        if order_by not in orderBy:
+            return None
+        
+        field, asc = order_by.split(":")
+        field = field.strip().lower()
+        asc = asc.strip().lower()
+        if field == "location":
+            field = "approx_loc"
+        elif field == "map square #":
+            field = "map_square"
+        
+        asc = asc == 'ascending'
+        
+        return f'{"" if asc else "-"}{field}'
+
+    # Pulling the params from the request
     name = request.GET.get("name", None)
-    is_searching_by_name = name is not None and name.strip() != ""
-    if is_searching_by_name:
-        matching_photographers = (Photographer.objects.filter(name__icontains=name)
-                                                      .order_by("name")
-                                                      .prefetch_related("photo_set"))
+    location = request.GET.get("location", None)
+    map_square = request.GET.get("square", None)
+    name_start = request.GET.get("name_starts_with", None)
+    order_by = request.GET.get("order_by", None)
+
+    # Pagination params
+    page_number = request.GET.get("page", None)
+    count_per_page = 50 
+
+    search_params = {}
+    
+    # Validating and adding all of the params
+    if name is not None and name.strip() != "":
+        search_params["name__icontains"] = name
+    if location is not None and location.strip() != "" and location in locations:
```

Comment:
A bunch of this validation is a little over the top -- e.g., we're sending the locations list to the frontend and populating the request only from those locations, so we don't really need guardrails checking the value is still in locations. (Same with nameStartsWith, etc.)